code/controllers/subsystem/blackbox.dm

```diff
@@ -187,7 +187,7 @@ Versioning
 						"gun_fired" = 2)
 */
 /datum/controller/subsystem/blackbox/proc/record_feedback(key_type, key, increment, data, overwrite)
-	if(sealed || !key_type || !istext(key) || !isnum(increment) || !data)
+	if(sealed || !key_type || !istext(key) || !isnum(increment || !data))

```

Comment:
U sure bout dat?

Reply:
?

Reply:
!data... inside isnum

Reply:
It logically resolves to `IF increment IS NOT num OR IF data IS false` because byond

Reply:
````
/world/New()
	dothething(4, 6)
	dothething(4, "bob")
	dothething("bob", 6)
	dothething("bob", "steve")
	dothething(4)
	dothething(data = 4)
	dothething("bob")

proc/dothething(increment, data)
	world.log << "---------------"
	if(!increment)
		world.log << "no increment"
	else
		world.log << "is increment"
	if(!data)
		world.log << "no data"
	else
		world.log << "is data"
	if(!isnum(increment))
		world.log << "increment not num"
	else
		world.log << "increment num"
	if(!isnum(data))
		world.log << "data not num"
	else
		world.log << "data num"
	if(!isnum(increment || !data))
		world.log << "not isnum increment or data"
	else
		world.log << "is num increment or data"
````
````
---------------
is increment
is data
increment num
data num
is num increment or data
---------------
is increment
is data
increment num
data not num
is num increment or data
---------------
is increment
is data
increment not num
data num
not isnum increment or data
---------------
is increment
is data
increment not num
data not num
not isnum increment or data
---------------
is increment
no data
increment num
data not num
is num increment or data
---------------
no increment
is data
increment not num
data num
is num increment or data
---------------
is increment
no data
increment not num
data not num
not isnum increment or data
````